### ISSUE TITLE
JS interop broken in flutter main

### ISSUE THREAD

**Issue by mit-mit** [dart-lang/sdk#51998]
When trying to run me small test app I get:
```
Rejecting promise with error: Expected a value of type 'FontCollectionNamespace', but got one of type 'Null'
_js_helper.TypeErrorImpl.new
the Dart compiler exited unexpectedly.
```

This is with Flutter main tip-of-tree, which currently has Dart `3.0.0-417.0.dev`. If I roll back to a week ago (Dart SDK `3.0.0-261.0.dev) then it works as expected.

**Comment by mit-mit** [comment]
cc @srujzs

**Comment by srujzs** [comment]
I think this was a cache invalidation issue, see https://github.com/flutter/flutter/issues/123784.

**Comment by mit-mit** [comment]
Yes, can confirm that `rm -rf bin/cache` resolved the JS interop issue so resolving this.

**Comment by mit-mit** [comment]
It looks like https://github.com/flutter/flutter/issues/123784 was closed as a dupe of the present issue, but I just hit it again in https://github.com/dart-lang/sdk/issues/51998

I think we have to fix this for 3.10!
